### PR TITLE
Pending Ticket Auto Close 1.9.x

### PR DESCRIPTION
**Pull request by jdelhome3578**
This will automatically close Pending Status tickets based on a database value entered on the ticket settings page.
To do:
- [ ] remove pending ticket queue and stats after #2577 implementation

Need help:
- [ ] set status using status name instead of id

### CODE REVIEW

**Comment by infectormp** [comment]
:+1:

**Comment by infectormp** [comment]
You use new status Pending, but why you don't use Resolved status?

**Comment by jdelhome3578** [comment]
We were using Pending way before resolved ever became.

**Comment by infectormp** [comment]
Maybe better make status configurable?

**Comment by jdelhome3578** [comment]
Maybe @greezybacon or @protich have some ideas on that.

**Comment by ntozier** [comment]
I'm biased [since I wrote the Pending/autoclose for 1.6/1.7] but I think that this is a long over due feature.

Pending to me means that its pending customer response (and may or may not be resolved)
Resolved to me means that the issue is resolved.
Or at least that's how we have used them here since 1.6.

**Comment by jdelhome3578** [comment]
We use this Pending status when we are waiting for a response to confirm a fix has worked. If they don't respond this closes the ticket for us.

**Comment by jdelhome3578** [comment]
@greezybacon @protich Would this be something to implement in 1.10 or a plugin?

**Comment by jdelhome3578** [comment]
When custom queues #2577 is merged this code can be reduced.

File: include/class.ticket.php [hunk 1]
```diff
@@ -3019,5 +3026,26 @@ function checkOverdue() {
...
+	if($grace == 1){$dayplural = 'day';} else {$dayplural = 'days';}
+
+	// select all tickets marked as 'pending' where updated is older than ($days) days ago
+	$sql  = 'SELECT ticket_id FROM ' .TICKET_TABLE .' ticket '
+		 .' INNER JOIN '.TICKET_STATUS_TABLE.' status ON (status.id=ticket.status_id AND status.name = "pending") '
+	     .' AND TIME_TO_SEC(TIMEDIFF(NOW(),ticket.updated))>='.$grace.'*86400';
+
+        if(($res=db_query($sql)) && db_num_rows($res)) {
+            while(list($id)=db_fetch_row($res)) {
+                if(($ticket=Ticket::lookup($id)) && $ticket->setStatus('3', 'Ticket Closed by the SYSTEM after '.$grace.' '.$dayplural.' of no activity.', false));
```
Comment (jdelhome3578) [inline include/class.ticket.php]: @greezybacon I am unable to get this to work with `setStatus('closed', ...` it looks like it should work based off of ` function setState($state, $alerts=false) {
        switch(strtolower($state)) {
            case 'open':
                return $this->setStatus('open');
                break;
            case 'closed':
                return $this->setStatus('closed');
                break;`

**Comment by jdelhome3578** [comment]
@ntozier any ideas on the code comments

**Comment by jdelhome3578** [comment]
@greezybacon can you provide any assistance with the `setstatus()` function?

**Comment by jdelhome3578** [comment]
> Maybe better make status configurable?

@infectormp Made an update in #2865 with your suggestion.

**Comment by jdelhome3578** [comment]
For older version 1.9.x 
see PR #2917 for 1.10 version

File: include/class.ticket.php (hunk 1 above)
Comment (mckaygerhard) [inline include/class.ticket.php]: any feedback bout use the setState ?

**Comment by mckaygerhard** [comment]
any feedback about working or not?

**Pull request closed without merging**